MobileCenter/MobileCenter/Internals/Storage/MSLogDBStorage.m

```diff
@@ -0,0 +1,253 @@
+  }
+
+  // Generate batch Id.
+  logsAvailable = logEntries.count > 0;
+  if (logsAvailable) {
+    batchId = MS_UUID_STRING;
+    [self.batches setObject:dbIds forKey:[groupId stringByAppendingString:batchId]];
+  }
+
+  // Load completed.
+  if (completion) {
+    completion(logs, batchId);
+  }
+
+  // Return YES if more logs available.
+  return moreLogsAvailable;
+}
+
+#pragma mark - Delete logs
+
+- (NSArray<id<MSLog>> *)deleteLogsWithGroupId:(NSString *)groupId {
+  NSArray<id<MSLog>> *logs = [self logsFromDBWithGroupId:groupId];
+
+  // Delete logs
+  [self deleteLogsFromDBWithColumnValue:groupId columnName:kMSGroupIdColumnName];
+
+  // Delete related batches.
+  for (NSString *batchKey in [self.batches allKeys]) {
+    if ([batchKey hasPrefix:groupId]) {
+      [self.batches removeObjectForKey:batchKey];
+    }
+  }
+  return logs;
+}
+
+- (void)deleteLogsWithBatchId:(NSString *)batchId groupId:(NSString *)groupId {
+
+  // Get log Ids.
+  NSString *batchIdKey = [groupId stringByAppendingString:batchId];
+  NSArray<NSNumber *> *Ids = self.batches[batchIdKey];
+
+  // Delete logs and associated batch.
+  if (Ids.count > 0) {
+    [self deleteLogsFromDBWithColumnValues:Ids columnName:kMSIdColumnName];
+    [self.batches removeObjectForKey:batchIdKey];
+  }
+}
+
+#pragma mark - DB selection
+
+- (NSArray<id<MSLog>> *)logsFromDBWithGroupId:(NSString *)groupId {
+
+  // Get log entries for the given group Id.
+  NSString *condition = [NSString stringWithFormat:@"\"%@\" = '%@'", kMSGroupIdColumnName, groupId];
+  NSArray<NSArray *> *logEntries = [self logsFromDBWhere:condition];
+
+  // Get logs only.
+  NSMutableArray<id<MSLog>> *logs = [NSMutableArray<id<MSLog>> new];
+  for (NSArray *logEntry in logEntries) {
+    [logs addObject:logEntry[self.logColumnIndex]];
+  }
+  return logs;
+}
+
+- (NSArray<NSArray *> *)logsFromDBWhere:(NSString *_Nullable)condition {
+  NSMutableArray<NSArray *> *logEntries = [NSMutableArray<NSArray *> new];
+  NSMutableString *query = [NSMutableString stringWithFormat:@"SELECT * FROM \"%@\"", kMSLogTableName];
+  if (condition.length > 0) {
+    [query appendFormat:@" WHERE %@", condition];
+  }
+  NSArray<NSArray *> *entries = [self executeSelectionQuery:query];
+
+  // Get logs from DB.
+  for (NSMutableArray *row in entries) {
+    NSNumber *dbId = row[self.idColumnIndex];
+    NSData *logData = [[NSData alloc] initWithBase64EncodedString:row[self.logColumnIndex]
+                                                          options:NSDataBase64DecodingIgnoreUnknownCharacters];
+    id<MSLog> log;
+
+    // Deserialize the log.
+    @try {
+      log = [NSKeyedUnarchiver unarchiveObjectWithData:logData];
+    } @catch (NSException *exception) {
+
+      // The archived log is not valid.
+      MSLogError([MSMobileCenter logTag], @"Deserialization failed for log with Id %@: %@", dbId, exception);
+      [self deleteLogFromDBWithId:dbId];
+      continue;
+    }
+
+    // Update with deserialized log.
```

Comment:
I see your point :blush:, though there is one advantage in using `continue`, we are sure that nothing else will run after it. It makes it future proof in a sens that we don't have to recall ourselves that we have to not put anything after the catch block or we get some inconsistencies.